### ISSUE TITLE
WebException in the callbackfrom github

### ISSUE THREAD

**Issue by giuliohome** [Tarmil/websharper.oauth.sample#1]
When I build the sample with the correct app id/secret, after I authorize the login from github, I see the following error in the callback

`Error: System.Net.WebException: The request was aborted: Could not create SSL/TLS secure channel.. in System.Net.HttpWebRequest.GetRequestStream(TransportContext& context) in System.Net.HttpWebRequest.GetRequestStream() in WebSharper.OAuth.OAuth2.GetAccessToken@154.Invoke(Unit unitVar) in Microsoft.FSharp.Control.AsyncBuilderImpl.callA@839.Invoke(AsyncParams`1 args)`

**Comment by giuliohome** [comment]
Off topic

This has nothing to do with the issue, but just out of my curiosity about the terminology in the main fs and html for the ws holes. Are the  names  `PrivateNotLoggedInContent` and `PrivateLoggedInContent` accidentally used one in place of the other?

**Comment by Tarmil** [comment]
Hi, thanks for the report! This should be fixable with a call to this during app startup:

```fsharp
ServicePointManager.SecurityProtocol <- SecurityProtocolType.Tls12
```

I'll check that it does indeed fix the issue and push a fix.

**Comment by giuliohome** [comment]
When I try to do that in my local project, I get an `ArgumentNullException`:
```
[Value cannot be null. Parameter name: value 
Parameter name:text]
   System.Web.UI.HtmlTextWriter.WriteEncodedText(String text) +2936566
   WebSharper.UI.Templating.Runtime.writeStringParts@250-1.Invoke(StringPart _arg2) +556
   WebSharper.UI.Templating.Runtime.writeStringParts@250.Invoke(StringPart[] text, HtmlTextWriter w) +254
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +1692
   WebSharper.UI.Templating.Runtime.writeElement@293.Invoke(String tag, Attr[] attrs, FSharpOption`1 dataVar, Node[] children) +664
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +983
   WebSharper.UI.Templating.Runtime.writeElement@293.Invoke(String tag, Attr[] attrs, FSharpOption`1 dataVar, Node[] children) +664
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +983
   WebSharper.UI.Templating.Runtime.writeTemplate@249.Invoke(Template template, Boolean plain, FSharpList`1 extraAttrs, RenderContext ctx) +502
   WebSharper.UI.Templating.Runtime.write@381.Invoke(FSharpList`1 extraAttrs, Context ctx, HtmlTextWriter w, Boolean r) +506
   WebSharper.UI.Elt.Write(Context ctx, HtmlTextWriter h, Boolean res) +128
   WebSharper.UI.Server.-ctor@100-15.Invoke(FSharpList`1 a, Context ctx, HtmlTextWriter w, Boolean x) +71
   WebSharper.UI.Server.-ctor@95-14.Invoke(FSharpList`1 a, Context ctx, HtmlTextWriter w, FSharpOption`1 _arg1) +78
   WebSharper.UI.Elt.Write(Context ctx, HtmlTextWriter h, FSharpOption`1 res) +79
   <StartupCode$WebSharper-UI>.Write@68-1.Invoke(Doc d) +65
   Microsoft.FSharp.Primitives.Basics.List.iter(FSharpFunc`2 f, FSharpList`1 x) +26
   Microsoft.FSharp.Collections.ListModule.Iterate(FSharpFunc`2 action, FSharpList`1 list) +48
   WebSharper.UI.ConcreteDoc.Write(Context ctx, HtmlTextWriter w, FSharpOption`1 res) +255
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +3018
   WebSharper.UI.Templating.Runtime.writeElement@293.Invoke(String tag, Attr[] attrs, FSharpOption`1 dataVar, Node[] children) +664
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +983
   WebSharper.UI.Templating.Runtime.writeElement@293.Invoke(String tag, Attr[] attrs, FSharpOption`1 dataVar, Node[] children) +664
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +983
   WebSharper.UI.Templating.Runtime.writeElement@293.Invoke(String tag, Attr[] attrs, FSharpOption`1 dataVar, Node[] children) +664
   WebSharper.UI.Templating.Runtime.writeNode@309.Invoke(Node _arg5) +983
   WebSharper.UI.Templating.Runtime.writeTemplate@249.Invoke(Template template, Boolean plain, FSharpList`1 extraAttrs, RenderContext ctx) +502
   WebSharper.UI.Templating.Runtime.write@381.Invoke(FSharpList`1 extraAttrs, Context ctx, HtmlTextWriter w, Boolean r) +506
   WebSharper.UI.Server.TemplateDoc.Write(Context ctx, HtmlTextWriter h, Boolean res) +67
   WebSharper.UI.Server.-ctor@84-13.Invoke(Context ctx, HtmlTextWriter w, Boolean x) +60
   WebSharper.UI.Server.TemplateDoc.Write(Context ctx, HtmlTextWriter h, Boolean res) +67
   <StartupCode$WebSharper-UI>.$Doc.Page@42-1.Invoke(Stream s) +203
   WebSharper.Sitelets.respond@159.Invoke(Response response) +272
   Microsoft.FSharp.Control.args@823-1.Invoke(a a) +63
   Microsoft.FSharp.Control.AsyncIAsyncResult`1.GetResult() +356
   Microsoft.FSharp.Control.AsBeginEndHelpers.endAction(IAsyncResult iar) +64
   <StartupCode$FSharp-Core>.AsBeginEnd@1995.Invoke(IAsyncResult iar) +40
   WebSharper.Sitelets.HttpHandler.System-Web-IHttpAsyncHandler-EndProcessRequest(IAsyncResult result) +13
   System.Web.CallHandlerExecutionStep.System.Web.HttpApplication.IExecutionStep.Execute() +9748493
   System.Web.HttpApplication.ExecuteStepImpl(IExecutionStep step) +48
   System.Web.HttpApplication.ExecuteStep(IExecutionStep step, Boolean& completedSynchronously) +159
```

**Comment by Tarmil** [comment]
This looks like a separate issue, although I don't know what would cause it.

I pushed the above fix and it's working for me -- you can check the deploy at http://websharperoauthexample.azurewebsites.net/.

**Comment by giuliohome** [comment]
I'm getting a Runtime Error from the above url.

> Description: An application error occurred on the server. The current custom error settings for this application prevent the details of the application error from being viewed remotely (for security reasons). It could, however, be viewed by browsers running on the local server machine.

**Comment by Tarmil** [comment]
I think I found the issue: you don't have a display name configured on your GitHub account, so the `name` field is null, and the app didn't consider that case. This should be fixed by 905d7f41bf568ddbac30c8e05f36144e6bfa4617.

**Comment by giuliohome** [comment]
**Excellent !
Thank you very much!!!**
